### PR TITLE
refactor(plugin-mercury): remove the first ping and use buffer_state as the auth success indicator

### PR DESCRIPTION
**Pull request by ianwremmel**

### CODE REVIEW

File: packages/plugin-mercury/src/socket/socket-base.js [hunk 1]
```diff
@@ -341,29 +341,14 @@ export default class Socket extends EventEmitter {
...
-          /* istanbul ignore next */
-          return reject(error);
-        }
-      });
-
-      this.send({
-        id,
-        type: `ping`
-      });
+      };
```
Comment (bzang) [inline packages/plugin-mercury/src/socket/socket-base.js]: Should we have a rejected promise in if the wrong thing comes back?

File: packages/test-helper-mock-web-socket/package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "version": "0.7.34",
   "description": "",
   "license": "MIT",
-  "main": "dist/index.js",
+  "main": "src/index.js",
```
Comment (bzang) [inline packages/test-helper-mock-web-socket/package.json]: shouldn't main be `dist`? or does this package never get built independently?

Comment (ianwremmel) [inline packages/test-helper-mock-web-socket/package.json]: whoops. will fix

File: packages/plugin-mercury/src/socket/socket-base.js (hunk 1 above)
Comment (ianwremmel) [inline packages/plugin-mercury/src/socket/socket-base.js]: nope. the ping here is just to kick off ping/pong. it's not longer relevant to prove we've authenticated successfully. _ping has it's own logic for addressing mismatched pongs

**Pull request merged**